zerver/tornado/event_queue.py

```diff
@@ -1197,26 +1190,26 @@ def maybe_enqueue_notifications_for_message_update(
         # without extending the UserMessage data model.
         return
 
-    if stream_push_notify or stream_email_notify:
+    if user_data.stream_push_notify or user_data.stream_email_notify:
         # Currently we assume that if this flag is set to True, then
         # the user already was notified about the earlier message,
         # so we short circuit.  We may handle this more rigorously
         # in the future by looking at something like an AlreadyNotified
         # model.
         return
 
-    idle = presence_idle or receiver_is_off_zulip(user_profile_id)
+    idle = presence_idle or receiver_is_off_zulip(user_data.id)
```

Comment:
This should be a follow-up, but should we rename this to `user_data.user_id`?  I think that'd feel more natural.